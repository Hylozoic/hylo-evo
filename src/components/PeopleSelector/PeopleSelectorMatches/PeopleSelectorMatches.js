import PropTypes from 'prop-types'
import React from 'react'

import PersonListItem from 'components/PersonListItem'
import './PeopleSelectorMatches.scss'

const { any, arrayOf, func, shape, string } = PropTypes

const personType = shape({
  id: any,
  name: string,
  avatarUrl: string,
  community: string
})

export default class PeopleSelectorMatches extends React.Component {
  static propTypes = {
    addParticipant: func,
    currentMatch: any,
    matches: arrayOf(personType)
  }

  render () {
    const { addParticipant, currentMatch, matches, setCurrentMatch } = this.props
    return <ul styleName='people-selector-matches'>
      {matches && matches.map(match =>
        <PersonListItem
          key={match.id}
          active={match.id === currentMatch}
          person={match}
          onClick={() => addParticipant(match.id)}
          onMouseOver={() => setCurrentMatch(match.id)} />)}
    </ul>
  }
}
