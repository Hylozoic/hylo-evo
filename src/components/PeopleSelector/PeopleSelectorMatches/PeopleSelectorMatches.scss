.people-selector-matches {
  composes: messages-section from 'components/MessageSection/MessageSection.scss';
  flex: 1;
  padding-top: $space-5x;
}
