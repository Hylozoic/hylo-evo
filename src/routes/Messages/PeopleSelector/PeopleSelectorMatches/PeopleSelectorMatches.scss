.people-selector-matches {
  composes: messages-section from '../../MessageSection/MessageSection.scss';
  flex: 1;
  padding-top: $space-5x;
}
