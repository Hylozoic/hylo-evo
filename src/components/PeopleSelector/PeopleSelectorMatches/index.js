import PeopleSelectorMatches from './PeopleSelectorMatches'
export default PeopleSelectorMatches
