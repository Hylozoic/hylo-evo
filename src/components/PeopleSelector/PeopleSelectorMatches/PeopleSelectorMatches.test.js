import { mount, shallow } from 'enzyme'
import { pick } from 'lodash/fp'
import React from 'react'

import PeopleSelectorMatches from './PeopleSelectorMatches'
import people from './PeopleSelectorMatches.test.json'

it('matches the last snapshot', () => {
  const matches = people.map(p => ({
    ...pick([ 'id', 'name', 'avatarUrl' ], p),
    community: p.memberships[0].community.name
  }))
  const wrapper = shallow(
    <PeopleSelectorMatches currentMatch={people[0].id} matches={matches} />
  )
  expect(wrapper).toMatchSnapshot()
})

it('calls addParticipant with correct id when item clicked', () => {
  const addParticipant = jest.fn()
  const matches = [ { id: '1' }, { id: '2' } ]
  const wrapper = mount(
    <PeopleSelectorMatches addParticipant={addParticipant} matches={matches} />
  )
  wrapper.find('li').first().simulate('click')
  expect(addParticipant).toBeCalledWith('1')
})

it('calls setCurrentMatch with correct id when item moused over', () => {
  const setCurrentMatch = jest.fn()
  const matches = [ { id: '1' }, { id: '2' } ]
  const wrapper = mount(
    <PeopleSelectorMatches setCurrentMatch={setCurrentMatch} matches={matches} />
  )
  wrapper.find('li').last().simulate('mouseOver')
  expect(setCurrentMatch).toBeCalledWith('2')
})
