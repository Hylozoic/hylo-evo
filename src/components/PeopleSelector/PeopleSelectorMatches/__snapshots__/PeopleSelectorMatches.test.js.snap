// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the last snapshot 1`] = `
<ul
  data-styleName="people-selector-matches"
>
  <PersonListItem
    active={true}
    onClick={[Function]}
    onMouseOver={[Function]}
    person={
      Object {
        "avatarUrl": "https://s3.amazonaws.com/uifaces/faces/twitter/saarabpreet/128.jpg",
        "community": "Associate",
        "id": "72297",
        "name": "Cristal Heller",
      }
    }
  />
  <PersonListItem
    active={false}
    onClick={[Function]}
    onMouseOver={[Function]}
    person={
      Object {
        "avatarUrl": "https://s3.amazonaws.com/uifaces/faces/twitter/orkuncaylar/128.jpg",
        "community": "Associate",
        "id": "72552",
        "name": "Helga Cronin",
      }
    }
  />
  <PersonListItem
    active={false}
    onClick={[Function]}
    onMouseOver={[Function]}
    person={
      Object {
        "avatarUrl": "https://s3.amazonaws.com/uifaces/faces/twitter/larrygerard/128.jpg",
        "community": "Associate",
        "id": "72257",
        "name": "Isadore Herzog",
      }
    }
  />
  <PersonListItem
    active={false}
    onClick={[Function]}
    onMouseOver={[Function]}
    person={
      Object {
        "avatarUrl": "https://s3.amazonaws.com/uifaces/faces/twitter/antonyzotov/128.jpg",
        "community": "Associate",
        "id": "72041",
        "name": "Oma Hettinger",
      }
    }
  />
</ul>
`;
